### ISSUE TITLE
The method 'findRenderObject' was called on null.

### ISSUE THREAD

**Issue by bbence84** [Afonsocraposo/buttons_tabbar#9]
I am a Flutter novice, so forgive me if I am doing somethign wrong. I am getting the following error when switching between tabs (all of which build a relatively complex grouped list view). The views show up no problem, but the log will have this error:

```
The following NoSuchMethodError was thrown while dispatching notifications for TabController:
The method 'findRenderObject' was called on null.
Receiver: null
Tried calling: findRenderObject()

When the exception was thrown, this was the stack: 
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _ButtonsTabBarState._scrollTo (package:buttons_tabbar/buttons_tabbar.dart:446:48)
#2      _ButtonsTabBarState._setCurrentIndex (package:buttons_tabbar/buttons_tabbar.dart:405:5)
#3      _ButtonsTabBarState._goToIndex (package:buttons_tabbar/buttons_tabbar.dart:394:7)
#4      _ButtonsTabBarState._handleController (package:buttons_tabbar/buttons_tabbar.dart:264:7)
...
The TabController sending notification was: Instance of 'TabController'

```

Code that builds the tabs:

```
  return DefaultTabController(
      length: 9,
      child: Column(
        children: <Widget>[
          ButtonsTabBar(
            contentPadding: EdgeInsets.all(1),
            backgroundColor: Theme.of(context).primaryColor,
            unselectedBackgroundColor: Colors.white,
            labelStyle: TextStyle(color: Colors.white),
            unselectedLabelStyle:
                TextStyle(color: Theme.of(context).primaryColor),
            radius: 50,
            tabs: [
              Tab(
                icon: Icon(
                  FontAwesomeIcons.xxxx,
                ),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(
                  FontAwesomeIcons.xxxx,
                ),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.xxxx),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.xxxx),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.xxxx),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.xxxx),
                text: "xxxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.xxxx),
                text: "xxxx",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.smile),
                text: "Viselkedés",
              ),
              Tab(
                icon: Icon(FontAwesomeIcons.questionCircle),
                text: "Egyéb",
              ),
            ],
          ),
          Expanded(
            child: TabBarView(
              children: <Widget>[
                buildListWithFilter(database, null),
                buildListWithFilter(database, 'xxxx'),
                buildListWithFilter(database, 'xxxxx'),
                buildListWithFilter(database, 'xxxxx'),
                buildListWithFilter(database, 'xxxx'),
                buildListWithFilter(database, 'xxxx'),
                buildListWithFilter(database, 'xxxx'),
                buildListWithFilter(database, 'xxxx'),
                buildListWithFilter(database, 'other'),
              ],
            ),
          ),
        ],
      ),
    );
  }



  StreamBuilder<List<Entry>> buildListWithFilter(
      Database database, String entryType) {
    return StreamBuilder<List<Entry>>(
        stream: database.entriesStream(entryType: entryType),
        builder: (context, snapshot) {
          return snapshot.hasData
              ? StickyGroupedListView<dynamic, String>(
                  elements: snapshot.data,
                  groupBy: (dynamic element) =>
                      DateFormat.yMd().format(element.date.toDate()).toString(),
                  separator: _buildSeparator(),
                  groupSeparatorBuilder: (dynamic element) =>
                      _buildGroupSeparator(context, element),
                  itemBuilder: (context, entry) => MainListTile(
                    entry: entry,
                  ),
                  order: StickyGroupedListOrder.DESC, // optional
                )
              : Container();
        });
  }
```

Any suggestions?

**Comment by Afonsocraposo** [comment]
I'll look into it ;)

**Comment by Afonsocraposo** [comment]
Alright, I just tested:
```dart
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: DefaultTabController(
          length: 9,
          child: Column(
            children: <Widget>[
              ButtonsTabBar(
                contentPadding: EdgeInsets.all(1),
                backgroundColor: Theme.of(context).primaryColor,
                unselectedBackgroundColor: Colors.white,
                labelStyle: TextStyle(color: Colors.white),
                unselectedLabelStyle:
                    TextStyle(color: Theme.of(context).primaryColor),
                radius: 50,
                tabs: [
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "xxxx",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "Viselkedés",
                  ),
                  Tab(
                    icon: Icon(Icons.home),
                    text: "Egyéb",
                  ),
                ],
              ),
              Expanded(
                child: TabBarView(
                  children: <Widget>[
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                    Container(),
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
```
Which is the barebones version since I do not have all the missing parts of your code. It didn't give me any error, so I'd say you're probably having problems with something not related with the `buttons_tabbar` package and it fails when integrated into it.
I'm sorry I can not give you a solution

**Comment by bbence84** [comment]
Thanks for checking so quickly! Actually I think there's an issue, but maybe indeed not with your package. 
But here is what I just found out:
1. With all 9 tabs, the tabs don't scroll automatically to the left when I click a rightmost tab. So I have to manually "drag" the tabs to the left for other tab buttons to appear
2. When I have kept only 3 tabs that fully are visible on the screen without having to automatically "scroll" for the control, it also does not show any errors. 
So it must be something with the fact that I have more tabs that can be displayed on the screen and autoscroll horizontally does not work.
Here is my Scaffold code, maybe that helps:


```
  Widget build(BuildContext context) {
    return Scaffold(
      floatingActionButton: FloatingActionButton(
        backgroundColor: Theme.of(context).primaryColor,
        child: Icon(Icons.add),
        onPressed: () => EditEntryPage.show(
          context,
          database: Provider.of<Database>(context, listen: false),
        ),
      ),
      body: SafeArea(child: _buildContentsGrouped(context)),
    );
  }
```

Additionally, I am using another package for bottom navigation bar, that uses a customer TabController. Maybe this interferes with your package? But again, only 3 tabs with your package works fine...

The bottom nav bar code:

```
class ConvexBottomBarScaffold extends StatelessWidget {
  const ConvexBottomBarScaffold({
    Key key,
    @required this.currentTab,
    @required this.onSelectTab,
    @required this.widgetBuilders,
    @required this.navigatorKeys,
    @required this.tabController,
  }) : super(key: key);

  final TabItem currentTab;
  final ValueChanged<TabItem> onSelectTab;
  final Map<TabItem, WidgetBuilder> widgetBuilders;
  final Map<TabItem, GlobalKey<NavigatorState>> navigatorKeys;
  final TabController tabController;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: TabBarView(
        controller: tabController,
        children: <Widget>[..._buildTabWidgetsList(context)],
      ),
      bottomNavigationBar: tab.ConvexAppBar(
          color: Colors.black,
          backgroundColor: Theme.of(context).primaryColor,
          controller: tabController,
          items: [
            _buildItem(context, TabItem.mainlist),
            _buildItem(context, TabItem.articles),
            _buildItem(context, TabItem.docfinder),
            _buildItem(context, TabItem.account),
          ],
          onTap: (index) => onSelectTab(TabItem.values[index])),
    );
  }

  tab.TabItem _buildItem(BuildContext context, TabItem tabItem) {
    final itemData = TabItemData.allTabs[tabItem];
    final color =
        currentTab == tabItem ? Theme.of(context).primaryColor : Colors.white;
    return tab.TabItem(
      icon: Icon(
        itemData.icon,
        color: color,
      ),
      title: itemData.title,
    );
  }

  _buildTabWidgetsList(BuildContext context) {
    final widgetList = <Widget>[];
    widgetBuilders.forEach((key, WidgetBuilder widgetBuilder) {
      widgetList.add(widgetBuilder(context));
    });
    return widgetList;
  }
}
```

**Comment by Afonsocraposo** [comment]
Hi! You were right. It was a problem on my side. Sorry for closing the issue so quickly.
Everything should be fixed now in version `1.0.3`. To use the fixed version, update your `pubspec.yaml` to:
```yaml
dependencies:
  ...
  buttons_tabbar: ^1.0.3
```

**Comment by bbence84** [comment]
Many many thanks for the lighting fast solution! :) Happy new year! :)
